manifest/v1alpha/metadata_annotations.go

```diff
@@ -0,0 +1,68 @@
+package v1alpha
+
+import (
+	"regexp"
+	"unicode/utf8"
+
+	"github.com/pkg/errors"
+
+	"github.com/nobl9/nobl9-go/internal/validation"
+)
+
+type (
+	MetadataAnnotations map[annotationKey]annotationValue
+
+	annotationKey   = string
+	annotationValue = string
+)
+
+func ValidationRuleMetadataAnnotations() validation.SingleRule[MetadataAnnotations] {
+	return validation.NewSingleRule(func(v MetadataAnnotations) error { return v.Validate() })
+}
+
+// Validate checks if the MetadataAnnotations keys and values are valid.
+func (a MetadataAnnotations) Validate() error {
+	for key, value := range a {
+		if err := a.validateKey(key); err != nil {
+			return err
+		}
+		if err := a.validateValue(key, value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+const (
+	minAnnotationKeyLength   = 1
+	maxAnnotationKeyLength   = 63
+	minAnnotationValueLength = 1
+	maxAnnotationValueLength = 1050
+)
+
+var (
+	annotationKeyRegexp = regexp.MustCompile(`^\p{L}([_\-0-9\p{L}]*[0-9\p{L}])?$`)
+)
+
+func (a MetadataAnnotations) validateKey(key annotationKey) error {
```

Comment:
I know we don't have map iterator support (yet) in `validation` pkg, but you can still define validation for each key dynamically and treat it as a separate object, that's how the map iterator would behave anyway, each key would be a new object as for JSON/YAML path there's no difference how you define that in code